indexer/indexer_tests/editable_map_object_test.cpp

```diff
@@ -275,54 +275,42 @@ UNIT_TEST(EditableMapObject_GetNamesDataSource)
   auto const namesDataSource = EditableMapObject::GetNamesDataSource(
       emo.GetNameMultilang(), nativeMwmLanguages, GetLangCode("ko"));
 
-  TEST_EQUAL(namesDataSource.names.size(), 9, ("All names except the default should be pushed into "
-                                           "data source plus empty mandatory names"));
-  TEST_EQUAL(namesDataSource.mandatoryNamesCount, 4,
-             ("Mandatory names count should be equal to Mwm languages + user`s language"));
-  TEST_EQUAL(namesDataSource.names[0].m_code, GetLangCode("de"),
-             ("German must be first because it is first in the list of languages for MWM"));
-  TEST_EQUAL(namesDataSource.names[1].m_code, GetLangCode("fr"),
-             ("French must be second because it is second in the list of languages for MWM"));
```

Comment:
Was this order lost? Why is it not important?

Reply:
The functionality changed and now only 'default_name' is always shown. This simplified the code and we now need fewer tests.

Reply:
If the country/region's languages are known, would it be more beneficial to also duplicate the default name into the tagged one, and encourage to add translations for those users who care? E.g. for Switzerland, when adding an object, I now see:
- French
- German
- (should also see Italian, need to fix it)
- English (always shown as the international language)
- Belarussian (my system language)

The last two in this case are less important and can be hidden by default (or left as is).

If I type German, then the name can be saved as name= and name:de=, right? (If fr, it, de are all changed, then there should be the sorted hard-coded order of languages per region, and the first one should be used as a name=).

By showing only "Name in local language" we can not be 100% sure which language was used to type it, right? And are losing the opportunity to save the name under an explicitly defined language.

That is true for adding new features on the map. For editing, we have different cases:
1. No name tag at all, and it can be edited by a user. Likely the same logic as above can be used.
2. Only default name= is present. Only it can be displayed.
3. A default name= and some translations are present. All of them should be displayed.

What would be more beneficial for OSM and users all over the world in the long term?

CC @Zverik

Reply:
The problem with displaying multilingual names too prominent ist that, at the moment, many people get confused by the two (or more) name fields and use them for short-name and long-name or invent translations where there are none (e.g. most Restaurants don't have multilingual names and adding e.g. name:en doesn't really make sense)

That's why this PR simplifies the UX to display just one name field by default.


The problem with name conversions is that it it only takes regions with only one language into account. So the idea doesn't work any more if there are two languages, like e.g. in Belgium. And keep in mind that the OM implementation of this idea never worked properly, resulting in wrong and nonsense data being uploaded to OSM.

That's why this PR removes name conversions.

So in the long term it ist most beneficial to fix the bugs of the OM editor. Then no wrong data is uploaded to OSM and users all over the world get higher quality map data.

Reply:
I agree that wrong data should not be uploaded to OSM. But is it the best way to fix that issue? The proposed approach is clear to every OSMer, but is it clear for inexperienced users?

One of the current issues is that users _don't read_ that the name field says "French", or "Dutch", and write there some unrelated stuff. Will they read and understand "As in the local language"? It's even more confusing than "French" or "Dutch".

If confusion comes from many fields, what about displaying only relevant map languages then? Isn't it important to understand which language was used by a user when the name was typed? Especially in 2+ official local languages (Belgium, Switzerland, Canada, etc.)?

Reply:
@map-per perhaps we could notify the user about what is "local language" is? Because in the current implementation, this may not be clear. So if I'm not a native speaker or just a tourist and cannot provide "name in local language" it would be better to skip this field? Because OM isn't for only experienced osmers, but for a wide range of users. What do you think?

Reply:
> One of the current issues is that users don't read that the name field says "French", or "Dutch", and write there some unrelated stuff. 

I have not noticed any such errors while reviewing OM edits. Are you sure they exist?


> Isn't it important to understand which language was used by a user when the name was typed? 

No, that's not important in the OSM tagging scheme. That's just the way OSM is, we can't change that.

> perhaps we could notify the user about what is "local language" is?

I think "As it is written in the local language" is clear enough

Reply:
> I have not noticed any such errors while reviewing OM edits. Are you sure they exist?

Didn't you explicitly mention that issue above? Or am I misunderstood you?

> The problem with displaying multilingual names too prominent is that, at the moment, many people get confused by the two (or more) name fields and use them for short-name and long-name or invent translations where there are none (e.g. most Restaurants don't have multilingual names and adding e.g. name:en doesn't really make sense)

Reply:
Ah, OK, then that's a misunderstanding. People invent translations, but they invent them in the right language.

Reply:
Now that I think about it again, the short-name and long-name issues I saw are probably rather a result of the name conversion. So the users probably added a german name that was saved to name=* and then had second thought and changed the german name, but OM didn't change name=* but added name:de=*.